bodhi-client/bodhi/client/oidcclient.py

```diff
@@ -149,6 +149,43 @@ def login(self):
             self._run_http_server()
         click.secho("Login successful!", fg="green")
 
+    def login_with_kerberos(self):
+        """Login to the OIDC provider using Kerberos.
+
+        How to test:
+        1. obtain a TGT via `kinit` command for your FAS account
+        2. `from bodhi.client.bindings import BodhiClient`
+        3. `bodhi = BodhiClient()`
+        4. `bodhi.login_with_kerberos()`
+        5. A file with tokens should exist: `~/.config/bodhi/client.json`
+
+        Raises:
+            OiDCClientError if there is a problem during the auth process
+        """
+        authorization_endpoint = self.metadata["authorization_endpoint"]
+        uri, state_ = self.client.create_authorization_url(authorization_endpoint)
+        response = requests.get(
+            uri,
+            auth=HTTPKerberosAuth(
+                # REQUIRED is not working with id.fedoraproject.org
+                mutual_authentication=OPTIONAL,
+            ),
+        )
+        response.raise_for_status()
+        try:
+            value = re.findall(
+                r"<title>\s*(code=[\w\-_=;&]+)\s*</title>", response.text
+            )[0]
+        except IndexError:
+            raise OIDCClientError(
+                f'Unable to locate OIDC code in the response from "{uri}".'
+            )
+        self.auth_callback(f"?{value}")
+        if not self.tokens:
+            raise OIDCClientError(
```

Comment:
Same here, please add a test case for this situation. What caused this in your tests?

Reply:
I'm actually going to drop this since the other login method doesn't do this check.

I don't remember honestly, it could also be a leftover from debugging :)